impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.internal.persistence;
+
+import org.ehcache.Cache;
+import org.ehcache.config.persistence.PersistenceConfiguration;
+import org.ehcache.spi.service.LocalPersistenceService;
+import org.ehcache.spi.service.ServiceConfiguration;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.channels.FileLock;
+
+/**
+ * @author Alex Snaps
+ */
+public class DefaultLocalPersistenceService implements LocalPersistenceService {
+
+  private final File rootDirectory;
+  private FileLock lock;
+  private File lockFile;
+
+  public DefaultLocalPersistenceService(final PersistenceConfiguration persistenceConfiguration) {
+    rootDirectory = persistenceConfiguration.getRootDirectory();
+  }
+
+  @Override
+  public synchronized void start(final ServiceConfiguration<?> config) {
+    createLocationIfRequiredAndVerify(rootDirectory);
+    try {
+      lockFile = new File(rootDirectory + File.separator + ".lock");
+      final RandomAccessFile rw = new RandomAccessFile(lockFile, "rw");
+      lock = rw.getChannel().lock();
+    } catch (IOException e) {
+      throw new RuntimeException("Couldn't lock rootDir: " + rootDirectory.getAbsolutePath(), e);
+    }
+  }
+
+  @Override
+  public synchronized void stop() {
+    try {
+      lock.release();
+      if (!lockFile.delete()) {
+        // todo log something?;
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Couldn't unlock rootDir: " + rootDirectory.getAbsolutePath(), e);
```

Comment:
Should we not simply log that instead of throwing? My thinking is that at this point, if we did not have any other IO exception earlier, it matters not much.